docs/transactions/source/transaction-validation.rst

```diff
@@ -1,34 +1,113 @@
+JSON Schema Validation is done by checking the transaction against
+a formal `JSON Schema  <http://json-schema.org/>`_
 defined in a set of `JSON Schema <http://json-schema.org/>`_ files.
 Those files can be found
 in the ``tx_schema/`` directory
-of the IPDB Transaction Spec repository.
+of the `IPDB Transaction Spec repository
+<https://github.com/ipdb/ipdb-tx-spec>`_.
 
-Tip 1: There's a nice explanation of JSON Schema in the website
-`"Understanding JSON Schema"
-<https://spacetelescope.github.io/understanding-json-schema/index.html>`_.
+.. note::
 
-Tip 2: Python 3 code for checking a transaction against those JSON Schema files
-can be found in the `source code of BigchainDB Server
-<https://github.com/bigchaindb/bigchaindb>`_.
-At the time of writing, it was in the file
-``bigchaindb/common/schema/__init__.py``.
+   Tip 1: There's a nice explanation of JSON Schema in the website
+   `"Understanding JSON Schema"
+   <https://spacetelescope.github.io/understanding-json-schema/index.html>`_.
+
+   Tip 2: Python 3 code for checking a transaction against JSON Schema files
+   can be found in the `source code of BigchainDB Server
+   <https://github.com/bigchaindb/bigchaindb>`_.
+   At the time of writing, it was in the file
+   ``bigchaindb/common/schema/__init__.py``.
 
 
 Other Constraints
 -----------------
 
-TODO
+The output.amount Rule
+^^^^^^^^^^^^^^^^^^^^^^
+
+For all outputs, once output.amount has been converted
+from a string to an integer,
+it must be between 1 and 9×10¹⁸, inclusive.
+The reason for the upper bound is to keep amount within what a server
+can comfortably represent using a 64-bit signed integer,
+i.e. 9×10¹⁸ is less than 2⁶³.
+
+The Duplicate Transaction Rule
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If a transaction is a duplicate of a previous transaction,
+then it's invalid.
+
+The Transaction ID Rule
+^^^^^^^^^^^^^^^^^^^^^^^
+
+The transaction ID (id) must be valid in that it must agree
+with the hash computed using the instructions given
+on :ref:`the page about transaction ID <Transaction ID>`.
+
+The TRANSFER Transaction Rules
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If a transaction is a TRANSFER transaction:
+
+#. If an input attempts to fulfill an output
+   that has already been fulfilled (i.e. spent or transferred)
+   by a previous valid transaction,
+   then the transaction is invalid.
+   (You don't have to check if the fulfillment string is valid.)
+#. If two or more inputs
+   (in the transaction being validated)
+   attempt to fulfill the *same* output,
+   then the transaction is invalid.
+   (You don't have to check any fulfillment strings.)
+#. The sum of the amounts on the inputs must equal
+   the sum of the amounts on the outputs.
+   In other words, a TRANSFER transaction can't create or destroy asset shares.
+#. For all inputs, if input.fulfills points to:
+    - a transaction that doesn't exist, then it's invalid.
+    - a transaction that's invalid, then it's invalid.
```

Comment:
We are not making assumptions on **how to store transactions**, so this point can be useful, but I'm a bit puzzled.

Reply:
That's why I added the parenthetical sentence, "(This check may be skipped if invalid transactions are never kept.)"